packages/astro/src/astro-components/unstyled/SignInButton.astro

```diff
@@ -0,0 +1,49 @@
+---
+import type { HTMLTag, Polymorphic } from 'astro/types'
+import type { SignInProps } from "@clerk/types";
+type Props<Tag extends HTMLTag> = Polymorphic<SignInProps & { as: Tag; id?: string; mode?: 'redirect' | 'modal' }>
```

Comment:
The [Polymorphic](https://docs.astro.build/en/guides/typescript/#polymorphic-type) type here is a helper that ensure that passed element is a valid HTML element.